x-pack/plugins/security_solution/public/cloud_security_posture/components/csp_details/insights_tab_csp.tsx

```diff
@@ -31,24 +32,35 @@ interface CspFlyoutPanels extends FlyoutPanels {
   left: CspFlyoutPanelProps;
 }
 
+// Type guard to check if the path is a PathPanel
+const isPathPanel = (path: PanelPath): path is PanelPath => {
+  return typeof path?.tab === 'string';
+};
+
 export const InsightsTabCsp = memo(
   ({ name, fieldName }: { name: string; fieldName: 'host.name' | 'user.name' }) => {
+    /* Using type cast here because originally I thought paths are part of Params when its actually its own field. The correct
+    way to solve this is by putting path outside params for every single openLeftPanel call in this folder, but given the time frame
+    We decided to do this for now */
     const panels = useExpandableFlyoutState() as CspFlyoutPanels;
-    const defaultTab = () =>
+
+    const getDefaultTab = () =>
       panels.left?.params?.hasMisconfigurationFindings
-        ? 'misconfigurationTabId'
+        ? CspInsightLeftPanelSubTab.MISCONFIGURATIONS
         : panels.left?.params?.hasVulnerabilitiesFindings
-        ? 'vulnerabilitiesTabId'
+        ? CspInsightLeftPanelSubTab.VULNERABILITIES
         : 'undefined';
```

Comment:
Note that returning the string `'undefined'` will cause falsy checks to evaluate to true. In the example below, the condition will never be met because `getDefaultTab()` always returns a truthy value. You’ll need to check for a specific string instead.

```ts
if (!getDefaultTab()) return <SomeDefaultComponent />;